src/recipe/error.rs

```diff
@@ -0,0 +1,354 @@
+        $crate::recipe::error::ParsingError {
+            src: $src.to_owned(),
+            span: $span,
+            label: None,
+            help: Some($help),
+            kind: $kind,
+        }
+    }};
+    ($src:expr, $span:expr, $kind:expr, label = $label:expr, help = $help:expr $(,)?) => {{
+        $crate::_error!($src, $span, $kind, $label, $help)
+    }};
+    ($src:expr, $span:expr, $kind:expr, help = $help:expr, label = $label:expr $(,)?) => {{
+        $crate::_error!($src, $span, $kind, $label, $help)
+    }};
+    ($src:expr, $span:expr, $kind:expr, $label:expr, $help:expr $(,)?) => {{
+        $crate::recipe::error::ParsingError {
+            src: $src.to_owned(),
+            span: $span,
+            label: Some($label),
+            help: Some($help),
+            kind: $kind,
+        }
+    }};
+}
+
+/// Macro to facilitate the creation of [`Error`]s.
+#[macro_export]
+#[doc(hidden)]
+macro_rules! _partialerror {
+    ($span:expr, $kind:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: None,
+            help: None,
+            kind: $kind,
+        }
+    }};
+    ($span:expr, $kind:expr, label = $label:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: Some($label),
+            help: None,
+            kind: $kind,
+        }
+    }};
+    ($span:expr, $kind:expr, help = $help:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: None,
+            help: Some($help),
+            kind: $kind,
+        }
+    }};
+    ($span:expr, $kind:expr, label = $label:expr, help = $help:expr $(,)?) => {{
+        $crate::_error!($src, $span, $kind, $label, $help)
+    }};
+    ($span:expr, $kind:expr, help = $help:expr, label = $label:expr $(,)?) => {{
+        $crate::_error!($src, $span, $kind, $label, $help)
+    }};
+    ($span:expr, $kind:expr, $label:expr, $help:expr $(,)?) => {{
+        $crate::recipe::error::PartialParsingError {
+            span: $span,
+            label: Some($label),
+            help: Some($help),
+            kind: $kind,
+        }
+    }};
+}
+
+/// Error handler for [`marked_yaml::LoadError`].
+pub(crate) fn load_error_handler(src: &str, err: marked_yaml::LoadError) -> ParsingError {
+    _error!(
+        src,
+        marker2span(src, marker(&err)),
+        ErrorKind::YamlParsing(err),
+        label = match err {
+            marked_yaml::LoadError::TopLevelMustBeMapping(_) => "expected a mapping here",
+            marked_yaml::LoadError::UnexpectedAnchor(_) => "unexpected anchor here",
+            marked_yaml::LoadError::UnexpectedTag(_) => "unexpected tag here",
+            _ => "here",
+        }
+    )
+}
+
+/// Convert a [`marked_yaml::Marker`] to a [`SourceSpan`].
+#[inline(always)]
+pub(crate) fn marker2span(src: &str, mark: marked_yaml::Marker) -> SourceSpan {
```

Comment:
This doesnt really seem to follow rust conventions. Maybe rename to `marker_to_span`?